gobblin-runtime/src/main/java/gobblin/runtime/mapreduce/GobblinOutputFormat.java

```diff
@@ -0,0 +1,151 @@
+import org.slf4j.LoggerFactory;
+
+import com.google.common.io.Closer;
+
+/**
+ * Hadoop {@link org.apache.hadoop.mapreduce.OutputFormat} implementation with a custom {@link OutputCommitter}
+ */
+public class GobblinOutputFormat extends NullOutputFormat<NullWritable, NullWritable> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(GobblinOutputFormat.class);
+
+  @Override
+  public OutputCommitter getOutputCommitter(TaskAttemptContext context) {
+    return new GobblinOutputCommitter();
+  }
+
+  /**
+   * Hadoop {@link OutputCommitter} implementation that overrides the default
+   * {@link #abortJob(JobContext, org.apache.hadoop.mapreduce.JobStatus.State)} behavior. This is necessary to add
+   * functionality for cleaning up staging data when the
+   * {@link com.linkedin.uif.runtime.JobLauncher#cancelJob(Properties)} method is called via Azkaban. Azkaban only
+   * allows the cancel method run to for 5 ms until it does a hard kill on the process, so instead, we will clean up the
+   * staging data in the AM.
+   */
+  private class GobblinOutputCommitter extends OutputCommitter {
+
+    @Override
+    public void abortJob(JobContext jobContext, JobStatus.State state) throws IOException {
+      LOG.info("Aborting Job: " + jobContext.getJobID());
+
+      Configuration conf = jobContext.getConfiguration();
+
+      URI fsUri = URI.create(conf.get(ConfigurationKeys.FS_URI_KEY, ConfigurationKeys.LOCAL_FS_URI));
+      FileSystem fs = FileSystem.get(fsUri, conf);
+
+      Path mrJobDir =
+          new Path(conf.get(ConfigurationKeys.MR_JOB_ROOT_DIR_KEY), conf.get(ConfigurationKeys.JOB_NAME_KEY));
+      Path jobInputDir = new Path(mrJobDir, "input");
+
+      for (FileStatus status : fs.listStatus(jobInputDir)) {
+
+        Closer workUnitFileCloser = Closer.create();
+        if (status.getPath().getName().endsWith(".wu")) {
+
+          WorkUnit wu = new WorkUnit();
+          wu.readFields(workUnitFileCloser.register(new DataInputStream(fs.open(status.getPath()))));
+          workUnitFileCloser.close();
+
+          LOG.info("Cleaning up staging data for WorkUnit: " + wu.getId());
+          int numBranches = wu.getPropAsInt(ConfigurationKeys.FORK_BRANCHES_KEY, 1);
+          for (int i = 0; i < numBranches; i++) {
+
+            String branchName = ForkOperatorUtils.getBranchName(wu, i, ConfigurationKeys.DEFAULT_FORK_BRANCH_NAME + i);
+
+            Path writerFilePath =
+                new Path(ForkOperatorUtils.getPathForBranch(wu.getExtract().getOutputFilePath(), branchName,
+                    numBranches));
+
+            String uri =
+                wu.getProp(ForkOperatorUtils.getPropertyNameForBranch(ConfigurationKeys.WRITER_FILE_SYSTEM_URI, i),
+                    ConfigurationKeys.LOCAL_FS_URI);
+
+            FileSystem wufs = FileSystem.get(URI.create(uri), new Configuration());
+
+            if (wu.contains(ForkOperatorUtils.getPropertyNameForBranch(ConfigurationKeys.WRITER_STAGING_DIR,
```

Comment:
Assign ForkOperatorUtils.getPropertyNameForBranch(ConfigurationKeys.WRITER_STAGING_DIR, numBranches, i) to a local variable so you don't call it twice.